packages/replay/src/session/Session.ts

```diff
@@ -24,59 +24,41 @@ interface SessionObject {
   segmentId: number;
 
   /**
-   * Is the session sampled? `null` if the sampled, otherwise, `session` or `error`
+   * The ID of the previous session.
+   * If this is empty, there was no previous session.
    */
-  sampled: Sampled;
-}
-
-export class Session {
-  /**
-   * Session ID
-   */
-  public readonly id: string;
+  previousSessionId?: string;
 
   /**
-   * Start time of current session
+   * Is the session sampled? `false` if not sampled, otherwise, `session` or `error`
    */
-  public started: number;
-
-  /**
-   * Last known activity of the session
-   */
-  public lastActivity: number;
-
-  /**
-   * Sequence ID specific to replay updates
-   */
-  public segmentId: number;
+  sampled: Sampled;
+}
 
-  /**
-   * Previous session ID
-   */
-  public previousSessionId: string | undefined;
+/**
+ * Get a session with defaults & applied sampling.
+ */
+export function makeSession(session: Partial<Session>, { sessionSampleRate, errorSampleRate }: SampleRates): Session {
```

Comment:
nit: If we get rid of the obj arg in favour of direct args with `sessionSampleRate` and `errorSampleRate` we'll save some bytes because we can minify those variables.

This makes sense to do if `makeSession` is not user facing, as then we can change the method signature in the future.